src/eventEmitter/index.js

```diff
@@ -0,0 +1,165 @@
+var
+  uuid = require('uuid');
+
+function EventEmitter(eventTimeout) {
+  Object.defineProperties(this, {
+    eventListeners: {
+      value: {}
+    },
+    eventTimeout: {
+      value: eventTimeout || 200,
+      writeable: false
+    }
+  });
+}
+
+/**
+ * Emit an event to all registered listeners
+ * An event cannot be emitted multiple times before a timeout has been reached.
+ */
+EventEmitter.prototype.emitEvent = function emitEvent(event) {
+  var
+    self = this,
+    now = Date.now(),
+    args = Array.prototype.slice.call(arguments, 1),
+    eventProperties = this.eventListeners[event];
+
+  if (!eventProperties) {
+    return false;
+  }
+
+  if (eventProperties.lastEmitted && eventProperties.lastEmitted >= now - this.eventTimeout) {
```

Comment:
After discussion, one option could be nice: to use a library that mimics node `events` methods (`on`, `off`, `emit` etc). 
When running on node.js, the sdk could just use the native module and use the polyfill when targeting the browser.